server/db/models/index.js

```diff
@@ -14,21 +15,26 @@ const db = require('../db')
 Category.hasMany(Product)
 
 // Order-Product M:M
-const OrdersProduct = db.define('OrdersProduct')
+const OrdersProduct = db.define('OrdersProduct', {
+  quantity: {
+    type: Sequelize.INTEGER,
+    allowNull: false,
+    validate: {min: 0},
+  },
+})
 Order.belongsToMany(Product, {through: OrdersProduct})
 Product.belongsToMany(Order, {through: OrdersProduct})
 
-// Cart (Product-User) M:M
-const Cart = db.define('Cart')
-Product.belongsToMany(User, {through: Cart})
-User.belongsToMany(Product, {through: Cart})
-
 // User-Address 1:M
 User.hasMany(Address)
 
 // User-Order 1:M
 User.hasMany(Order)
 
+// Order-Address 1:1
+Address.hasOne(Order, {foreignKey: 'shippingID'})
+Address.hasOne(Order, {foreignKey: 'billingID'})
+
```

Comment:
Should these associations be paired?
_ref: https://sequelize.org/master/manual/assocs.html#why-associations-are-defined-in-pairs-_
i.e. User.hasMany(Address) and Address.hasOne(User)

I don't think it's required, but it would ensure Address and Order have getUser methods on them.

Also should the Order-Address relationship be 1:m?
Address can belong to many orders

Reply:
The association pairs is a good idea. That would look like this:

```suggestion
// User-Address 1:M
User.hasMany(Address)
Address.BelongsTo(User)

// User-Order 1:M
User.hasMany(Order)
Order.belongsTo(User)
```

For Order-Address, this seems to work:
```suggestion
// Order-Address 1:M

Address.hasMany(Order, {foreignKey: 'billingID'})
Address.hasMany(Order, {foreignKey: 'shippingID'})
```
![Screen Shot 2020-09-23 at 4 58 10 PM](https://user-images.githubusercontent.com/24929052/94069046-f9370400-fdbd-11ea-80cc-1cba85fa8119.png)

If address belongs to many order we get the following. billingId and shippingId in address table
But the Order.hasOne(Address) creates an additional addressId column

```suggestion
// Order-Address 1:M

Address.belongsTo(Order, {foreignKey: 'shippingID'})
Address.belongsTo(Order, {foreignKey: 'billingID'})
Order.HasOne(Address)
```
![Screen Shot 2020-09-23 at 5 02 06 PM](https://user-images.githubusercontent.com/24929052/94069396-8417fe80-fdbe-11ea-9117-e094d4ca8a1f.png)

Reply:
User-Order and User-Address looks good!

For the Address-Order associations, did you try something like:
_ref: https://sequelize.org/master/manual/assocs.html#multiple-associations-involving-the-same-models_

```
Address.hasMany(Order, { as: 'BillingAddress', foreignKey: 'billingId' });
Address.hasMany(Order, { as: 'ShippingAddress', foreignKey: 'shippingId' });
Order.belongsTo(Address);
```